### DISCUSSION TITLE
Not able to connect to WSS link via RealtimeJS library

### DISCUSSION THREAD

**Discussion by asibilia** [supabase/supabase#7276] [Questions] [answered]
I'm struggling to find documentation on the exact steps required to use the Realtime JS library. I tried setting up my project similar to the `multiplayer` branch on `supabase/realtime` ([demo branch](https://github.com/supabase/realtime/tree/multiplayer/demo)) however it looks like they're using two separate keys for their clients... they've got a `REALTIME_ANON_KEY` and `SUPABASE_ANON_KEY`. Are these keys intended to be different? 

If I look at my API logs in the Supabase console I see the connection attempts coming in for 
``` 
{
 "url": "https://<project_id>.supabase.co/realtime/v1/websocket?apikey=<anon_key>&vsn=1.0.0"
}
```

Showing a `101` status code. However, when I look in my dev console on my local environment I see this warning

```
WebSocket connection to 'wss://<project_id>.supabase.co/realtime/v1/websocket?apikey=<anon_key>&vsn=1.0.0' failed: WebSocket is closed before the connection is established.
```

Anyone have suggestions?

** `<project_id>` and `<anon_key>` are placeholders 😉

**Comment by dshukertjr** [comment, marked as answer]
Hi @asibilia,

The easiest way to start using our realtime feature is by using the supabase-js library and not the realtime-js library. You can find the API reference [here](https://supabase.com/docs/reference/javascript/subscribe). 

The code under `multiplayer` branch is currently under development, and is not yet available to the public.
